src/lib/core/tt_hostname.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+enum {
+	TT_HOSTNAME_LEN_MAX = 63,
+	TT_HOSTNAME_SIZE_MAX = TT_HOSTNAME_LEN_MAX + 1,
+};
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/** DNS- and host-friendly name. */
+struct tt_hostname {
+	/** Name characters. */
+	char data[TT_HOSTNAME_SIZE_MAX];
+};
+
+/** Check if the string is a valid hostname. */
+bool
+tt_hostname_str_is_valid_n(const char *name, size_t name_len);
+
+static inline bool
+tt_hostname_str_is_valid(const char *name)
+{
+	return name != NULL && tt_hostname_str_is_valid_n(name, strlen(name));
```

Comment:
I'm not sure if this function will be used on unchecked user input or not, but I suggest to replace `strlen(name)` with `strnlen(name, TT_HOSTNAME_SIZE_MAX)`.

This way we won't have to introduce such a change once some static analysis tool warns that strlen might take too long on a large or not null-terminated string.

Reply:
Thanks, fixed.